ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitAmountsView.kt

```diff
@@ -0,0 +1,138 @@
+package org.p2p.uikit.components
+
+import android.content.Context
+import android.text.TextWatcher
+import android.util.AttributeSet
+import android.util.TypedValue
+import android.widget.EditText
+import android.widget.TextView
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.content.ContextCompat
+import androidx.core.widget.doAfterTextChanged
+import androidx.core.widget.doOnTextChanged
+import org.p2p.uikit.R
+import org.p2p.uikit.databinding.WidgetAmountsViewBinding
+import org.p2p.uikit.utils.inflateViewBinding
+import org.p2p.uikit.utils.showSoftKeyboard
+
+enum class FocusMode {
+    TOKEN,
+    CURRENCY
+}
+
+class UiKitAmountsView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : ConstraintLayout(context, attrs, defStyleAttr) {
+
+    private val binding = inflateViewBinding<WidgetAmountsViewBinding>()
+
+    private lateinit var tokenTextWatcher: TextWatcher
+    private lateinit var currencyTextWatcher: TextWatcher
+
+    var token: String? = null
+        set(value) {
+            field = value
+            binding.textViewToken.text = value
+        }
+
+    var currency: String? = null
+        set(value) {
+            field = value
+            binding.textViewCurrency.text = value
+        }
+
+    init {
+        background = ContextCompat.getDrawable(context, R.drawable.bg_amounts_view)
+
+        with(binding) {
+            val originalTextSize = editTextTokenAmount.textSize
+            editTextTokenAmount.doOnTextChanged { text, _, _, _ ->
+                handleAmountTextChanged(editTextTokenAmount, textViewTokenAutoSizeHelper, text, originalTextSize)
+            }
+            editTextCurrencyAmount.doOnTextChanged { text, _, _, _ ->
+                handleAmountTextChanged(editTextCurrencyAmount, textViewCurrencyAutoSizeHelper, text, originalTextSize)
+            }
+        }
+    }
+
+    fun getFocusMode(): FocusMode = if (binding.editTextTokenAmount.hasFocus()) {
```

Comment:
maybe FocusPlace then? if we return the place where Focus is

Reply:
What about `FocusField` then? Place is too ambiguous